.pre-commit-config.yaml

```diff
@@ -0,0 +1,61 @@
+default_language_version:
+  python: python3
+
+repos:
+  - repo: https://github.com/myint/autoflake
```

Comment:
I think it's better to use ruff
```
- repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.4.5
    hooks:
      - id: ruff
      - id: ruff-format
```
added this to pyproject for ruff
```
[tool.ruff]
preview = true
line-length = 120
fix = true
output-format = "grouped"

[tool.ruff.format]
exclude = [".git", ".venv", ".mypy_cache", ".tox", "__pycache__"]
```

then you can remove `autoflake` `isort` and `black`

Reply:
Also consider using mypy for typing.

```
- repo: https://github.com/pre-commit/mirrors-mypy
    rev: v1.10.0
    hooks:
      - id: mypy
        exclude: ^(tests/|examples/|docs/)
        additional_dependencies: [types-all]
```

and in pyproject.py
```
[tool.mypy]
check_untyped_defs = true
disallow_any_generics = true
disallow_incomplete_defs = true
disallow_untyped_defs = true
no_implicit_optional = true
show_error_codes = true
warn_unused_ignores = true
```